src/core/meshcop/commissioner.cpp

```diff
@@ -367,6 +372,22 @@ Error Commissioner::Stop(ResignMode aResignMode)
     return error;
 }
 
+Error Commissioner::SetId(const char *aId)
+{
+    Error   error = kErrorNone;
+    uint8_t len   = static_cast<uint8_t>(StringLength(aId, ot::MeshCoP::CommissionerIdTlv::kMaxLength));
+
+    // CommissionerIdTlv::SetCommissionerId trims the string to the maximum array size.
+    // Prevent this from happening returning an error.
+    VerifyOrExit((0 < len) && (len < mCommissionerId.kMaxLength), error = kErrorInvalidArgs);
```

Comment:
Is empty Id string (`len == 0`) invalid? 
The current `CommissionerIdTlv::SetCommissionerId()` implementation seem to accept it.